models/cow_protocol/ethereum/cow_protocol_ethereum_trades.sql

```diff
@@ -226,4 +226,8 @@ valued_trades as (
         ON trades.order_uid = efs.order_uid
 )
 
-select * from valued_trades
+select *,
+  -- Depending on the order kind (buy vs sell), the limit amount will always be equal to either the amount bought or sold.
+  -- We can therefore sum up the surplus relative to total buy and sell amount and extrapolate given the total trade's usd value.
+  ((limit_sell_amount - atoms_sold) / atoms_sold  * usd_value) + ((atoms_bought - limit_buy_amount) / atoms_bought * usd_value) as surplus_usd
```

Comment:
I wonder what happens here for partially-fillable orders... also couldn't we factor the `* usd_value` out of both terms here?
```suggestion
  usd_value * (((limit_sell_amount - atoms_sold) / atoms_sold) + ((atoms_bought - limit_buy_amount) / atoms_bought)) as surplus_usd
```

Please feel free to ignore the suggestion